dandi/upload.py

```diff
@@ -41,6 +41,7 @@ def upload(
     api_url = instance.api
 
     client = DandiAPIClient(api_url)
+    client.match_schema_version()
```

Comment:
+1 on doing it only in upload... I thought we might want it more generic (e.g. at DandiAPIClient level) but indeed there is no need.